File: MicroHawk/FactorialSum.cpp
#include<iostream>
using namespace std;

long long int fact(int number)
{
    if (number==0) return 1;
    else return number*fact(number-1);
}
void add(long long int prod)
{
    int sum=0;
    while(prod!=0)
    {
        sum=sum + prod%10;
        prod=prod/10;
    }
    cout<<sum<<endl;
    
}
int main() {
    // int n;
    // cin>>n;
    // for(int i=0;i<n;i++)
    // {
    int num;
    cin>>num;
    long long int prod=fact(num);
    add(prod);
    // }
    return 0;
}

File: Harshad/factorialSum.py
def fact(no):
    if no == 1:
        return 1
    else:
        return no * fact(no - 1)
def add_digit(number):
    type_cast = str(number)
    len_number = len(type_cast)
    ls_digits = []
    for i in range(len_number):
        ls_digits.append(int(type_cast[i]))

    summition = sum(ls_digits)
    return summition

def main():
    input_digit = int(input(">> "))
    actual_num = fact(input_digit)
    result = add_digit(actual_num)
    print(result)

main()
